unit_tests/test_aggregator.py

```diff
@@ -0,0 +1,415 @@
+                          return_value=test_response) as mocked_method:
+            aggregator_explicit.start_websocket_connection()
+            aggregator_explicit.execute_batch_commands()
+            mocked_method.assert_called_once()
+
+    @staticmethod
+    def test_execute_batch_commands__return_response(aggregator_explicit):
+        with patch.object(AggregatorWebsocketMessageReceiver, "wait_for_command_response",
+                          return_value=test_response):
+            aggregator_explicit.start_websocket_connection()
+            response = aggregator_explicit.execute_batch_commands()
+            assert response == test_response
+
+    @staticmethod
+    def test_execute_batch_commands__log_bid__offer_confirmation_called(aggregator_explicit):
+        with patch("gsy_e_sdk.aggregator.log_bid_offer_confirmation") as mocked_method:
+            aggregator_explicit.start_websocket_connection()
+            aggregator_explicit.execute_batch_commands()
+            mocked_method.assert_called()
+
+    @staticmethod
+    def test_execute_batch_commands__log_deleted_bid_offer_confirmation_called(
+            aggregator_explicit):
+        with patch("gsy_e_sdk.aggregator.log_deleted_bid_offer_confirmation") \
+                as mocked_method:
+            aggregator_explicit.start_websocket_connection()
+            aggregator_explicit.execute_batch_commands()
+            mocked_method.assert_called()
+
+
+@pytest.mark.parametrize("aggregator_specs", [test_agg, None])
+def test_list_aggregators(aggregator_specs, aggregator_explicit):
+    with patch("gsy_e_sdk.aggregator.blocking_get_request", return_value=aggregator_specs):
+        aggs_list = aggregator_explicit.list_aggregators()
+
+        if aggregator_specs is not None:
+            assert aggs_list == test_agg
+        else:
+            assert aggs_list == []
+
+
+def test_get_configuration_registry(aggregator_explicit):
+    conf = f"{TEST_CONFIGURATION_PREFIX}registry"
+
+    with patch("gsy_e_sdk.aggregator.blocking_get_request") as mocked_func:
+        aggregator_explicit.get_configuration_registry()
+        mocked_func.assert_called_with(conf, {}, "some_key")
+
+
+def test_delete_aggregator(aggregator_explicit):
+    conf = f"{TEST_AGGREGATOR_PREFIX}delete-aggregator/"
+
+    with patch("gsy_e_sdk.aggregator.blocking_post_request") as mocked_func:
+        aggregator_explicit.delete_aggregator()
+        mocked_func.assert_called_with(conf, {"aggregator_uuid": None}, "some_key")
+
+
+def test_add_to_batch_commands(aggregator_explicit):
+    # pylint: disable=protected-access
+    buffer_instance = aggregator_explicit.add_to_batch_commands
+    assert buffer_instance == aggregator_explicit._client_command_buffer
+
+
+def test_commands_buffer_length(aggregator_explicit):
+    patch("gsy_e_sdk.aggregator.ClientCommandBuffer.buffer_length",
+          new_callable=PropertyMock, return_value=3)
+    assert aggregator_explicit.commands_buffer_length == 3
```

Comment:
Here you are testing the `ClientCommandBuffer`. IMO you can delete this test from here and add it to the future unittests of the `ClientCommandBuffer`

Reply:
This test case may be a little tricky because the actual method (commands_buffer_length) just calls the `buffer_length` property of the `ClientCommandBuffer` and returns it. 
Therefore I'm patching this property and asserting that the return was as expected.

I'm not sure I understand why you say I'm testing the `ClientcommandBuffer` here though.

Reply:
I understand now, the `commands_buffer_length` is an Aggregator property. Ignore me.:-)